src/ucp/core/ucp_rkey.c

```diff
@@ -79,6 +102,15 @@ ucs_status_t ucp_ep_rkey_unpack(ucp_ep_h ep, void *rkey_buffer, ucp_rkey_h *rkey
 
     /* Read remote PD map */
     pd_map   = *(uint64_t*)p;
+
+    ucs_trace("unpacking rkey with pd_map 0x%"PRIx64, pd_map);
+
+    if (pd_map == 0) {
+        /* Dummy key return ok */
+        *rkey_p = &dummy_rkey;
```

Comment:
I've meant that there could be one global which is rkey buffer (u64 like you suggested), and another global for dummy rkey (the one you already added)